src/jsmn.c

```diff
@@ -188,6 +188,35 @@ static jsmnerr_t jsmn_parse_string(jsmn_parser *parser,
     return JSMN_ERROR_PART;
 }
 
+/**
+ * Decide how to handle string. It could be either Binary (base64) or String
+ */
+jsmnerr_t jsmn_handle_string(jsmn_parser *parser,
+                             CFStringInlineBuffer* buf, jsmntok_t *tokens,
+                             size_t num_tokens, CFIndex len, jsmntype_t type) {
+
+    if (type == JSMN_BINARY) {
+        if (CFStringGetCharacterFromInlineBuffer(buf, parser->pos + 1) != '\"') {
+            return JSMN_ERROR_INVAL;
+        }
+
+        parser->pos++;
+    }
+
+    jsmnerr_t r;
+
+#ifdef JSMN_DISABLED_ITEMS
+    r = jsmn_parse_string(parser, buf, tokens, num_tokens, len, type, parser->toksuper >= 0 ? !(tokens[parser->toksuper]._in_comment) : true);
+#else
+    r = jsmn_parse_string(parser, buf, tokens, num_tokens, len, type);
+#endif
+    if (r < 0) return r;
+    if (parser->toksuper != -1)
+        tokens[parser->toksuper].size++;
```

Comment:
I'd avoid `if` conditions without curly braces `{}` (I know they weren't there before…)

Harsh reminder of [Apple's goto fail bug](https://dwheeler.com/essays/apple-goto-fail.html) 😉 

And (also something that seems to have been broken before), the check `parser->toksuper` should only be there if there's `JSMN_PARENT_LINKS`.

```suggestion
    if (r < 0) {
        return r;
    }
#ifdef JSMN_PARENT_LINKS
    if (parser->toksuper != -1) {
      tokens[parser->toksuper].size++;
    }
#endif
```

Reply:
Okay. I haven't touched these parts, just refactored. I'll try to run all the tests and if all good, I'll apply and merge this pull requests. 🚀